snoop/management/commands/export-pgp.py

```diff
@@ -0,0 +1,84 @@
+from django.core.management.base import BaseCommand
+from pathlib import Path
+import base64
+from ... import models
+from ... import emails
+from ... import pgp
+from ...content_types import guess_content_type
+
+class Command(BaseCommand):
+    help = "Export digested pgp .eml files to a zip archive"
+
+    def add_arguments(self, parser):
+        parser.add_argument('where',
+                            help='SQL `where` clause to be ran ' +
+                                 'on the snoop_document table')
+        parser.add_argument('destination',
+                            help='path to the folder where ' +
+                                 'the files will be dumped')
+
+
+    def handle(self, where, destination, **options):
+        query = (
+            'SELECT id FROM snoop_document WHERE ' +
+            where.replace('%', '%%')
+        )
+        root = Path(destination)
+        done = 0
+        for doc in models.Document.objects.raw(query):
+            if emails.is_email(doc):
+                email = emails.open_email(doc)
+                if not email.pgp:
+                    print("id:", doc.id, "is not a pgp-encrypted email")
+                    continue
+                try:
+                    output = decrypt_email_file(email)
+                    dump_eml(root, doc.md5, output)
+                except Exception as e:
+                    print("id:", doc.id, "failed because of:\n", e)
+                else:
+                    print("id:", doc.id, "is done")
+                    done += 1
+            else:
+                print("id:", doc.id, "is not an email file")
+            print(done, "documents dumped.")
+
+
+def decrypt_email_file(email):
```

Comment:
Hey man, thanks a heap for writing this function, it would have taken me a while to piece the logic together on my own :)